spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/ExecutionCommandBuilder.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.deployer.spi.local;
+
+import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
+import org.springframework.core.io.Resource;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.springframework.cloud.deployer.spi.local.LocalDeployerProperties.LOCAL_PROPERTIES;
+
+/**
+ * A class to help create the execution command for launching a Java Process.
+ *
+ * @author Mark Pollack
+ */
+public class ExecutionCommandBuilder {
+
+    private static final String MAIN =  "main";
+
+    private static final String CLASSPATH = "classpath";
+
+    public void addJavaOptions(List<String> commands, Map<String, String> deploymentProperties,
+                               LocalDeployerProperties localDeployerProperties) {
+        String memory = null;
+        if (containsKey(deploymentProperties, "memory")) {
```

Comment:
This is part of the story that Thomas and I were working on, to add memory and cpu as top level properties.  We should merge this wrt to what we have now and then once Thomas' PR is ready, we can circle back to make the appropriate changes.  That is why I wanted this to get reviewed/merged early as possible wrt to coordinating that work.